### DISCUSSION TITLE
issue with `runtime.execute()`

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#409] [Q&A] [answered]
@pgundlach,

I have the following filter:

``` lua
runtime = require("runtime")

documento_pdf = runtime.variables['completefilename']:gsub("\\", "/")
ruta_pdf = runtime.variables['completepath']:gsub("\\", "/"):gsub('"',"/")
nombre_archivo = runtime.variables['filename']
extension = runtime.variables['extension']

runtime.execute({"sp", "--dummy", "--quiet", "--jobname", string.format("nuevo_%s%s",nombre_archivo,extension), "--filter", "spimpdok.lua", "--data", "spimpdok.xml", "-v", string.format("'completefilename=%s'",documento_pdf)})

print(documento_pdf)
```

Because of using `documento_pdf` inside `runtime.execute()`, I get the following error (although is printed fine after `runtime.execute()`):

```
filter.lua:3: attempt to index a non-table object(nil) with key 'gsub'
```

How can I access to the values of the `runtime.variables` table inside the `runtime.execute()` invocation?

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
How do you call the first sp run? The error message indicates that the variable `completefilename` is not set on invocation.

**Reply by pr-apes** [reply]
The value of `completefilename` is a filename (with full path included) in Windows.

Backslashes seem to be the problem here.

I guess it might be a bug (#411).

Many thanks for your help.

**Reply by pr-apes** [reply]
Since there is a fix for #411, I mark the question as answered.
